### PR TITLE
[new release] gluten, gluten-mirage, gluten-lwt, gluten-lwt-unix, gluten-eio and gluten-async (0.4.0)

### PR DESCRIPTION
**Pull request by anmonteiro**
A reusable runtime library for network protocols

- Project page: <a href="https://github.com/anmonteiro/gluten">https://github.com/anmonteiro/gluten</a>

##### CHANGES:

- gluten-eio: Add `gluten-eio` package, a gluten backend for
  [eio](https://github.com/ocaml-multicore/eio)
  ([anmonteiro/gluten#35](https://github.com/anmonteiro/gluten/pull/35))
- gluten-async: Allow connecting to a UNIX domain socket
  ([anmonteiro/gluten#40](https://github.com/anmonteiro/gluten/pull/40))
- gluten-async: Fix memory leak in the SSL / TLS implementations
  ([anmonteiro/gluten#48](https://github.com/anmonteiro/gluten/pull/48))

### CODE REVIEW

**Comment by anmonteiro** [comment]
There are some weird git exceptions unrelated to this package:

> fatal: Unable to create '/var/cache/obuilder/ocluster/git/opam-repository.git-b322f7b18d276db68487afc3dfc5a5befdd45818/.git/index.lock': File exists.


Not sure how to restart a CI run.

File: packages/gluten-async/gluten-async.0.4.0/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.08.0"}
+  "gluten" {= version}
+  "faraday-async" {>= "0.7.3"}
+  "async" {>= "v0.15.0"}
+  "core" {>= "v0.15.0"}
+]
+depopts: ["async_ssl" "tls-async"]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/gluten-async/gluten-async.0.4.0/opam:18]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/gluten-eio/gluten-eio.0.4.0/opam [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+homepage: "https://github.com/anmonteiro/gluten"
+bug-reports: "https://github.com/anmonteiro/gluten/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.08.0"}
+  "gluten" {= version}
+  "eio" {>= "0.7"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/gluten-eio/gluten-eio.0.4.0/opam:15]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/gluten-lwt-unix/gluten-lwt-unix.0.4.0/opam [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  "ocaml" {>= "4.08.0"}
+  "gluten-lwt" {= version}
+  "faraday-lwt-unix" {>= "0.7.3"}
+]
+depopts: [
+  "lwt_ssl" {>= "1.2.0"}
+  "tls"
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/gluten-lwt-unix/gluten-lwt-unix.0.4.0/opam:19]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/gluten-lwt/gluten-lwt.0.4.0/opam [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+homepage: "https://github.com/anmonteiro/gluten"
+bug-reports: "https://github.com/anmonteiro/gluten/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.08.0"}
+  "gluten" {= version}
+  "lwt" {>= "5.1.1"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/gluten-lwt/gluten-lwt.0.4.0/opam:15]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/gluten-mirage/gluten-mirage.0.4.0/opam [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.08.0"}
+  "gluten-lwt" {= version}
+  "faraday-lwt" {>= "0.7.3"}
+  "conduit-mirage" {>= "2.0.2"}
+  "mirage-flow" {>= "2.0.0"}
+  "cstruct" {>= "6.0.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/gluten-mirage/gluten-mirage.0.4.0/opam:18]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/gluten/gluten.0.4.0/opam [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+bug-reports: "https://github.com/anmonteiro/gluten/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.08.0"}
+  "bigstringaf" {>= "0.4.0"}
+  "faraday" {>= "0.7.3"}
+  "ke" {>= "0.5"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/gluten/gluten.0.4.0/opam:18]:
```suggestion
  ["dune" "subst"] {dev}
```

**Comment by mseri** [comment]
Thanks

**Pull request merged**